nested/blobStorageAccount.json

```diff
@@ -0,0 +1,58 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+            "metadata": {
+                "description": "Common Moodle values"
+            },
+            "type": "object"
+        }
+    },
+    "resources": [
+        {
+            "kind": "BlobStorage",
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('moodleCommon').blobStorageAccountName]",
+            "apiVersion": "2017-06-01",
+            "sku": {
+                "name": "[parameters('moodleCommon').blobStorageAccountType]"
+            },
+            "properties": {
+                "accessTier": "Hot",
+                "encryption": {
+                    "keySource": "Microsoft.Storage",
+                    "services": {
+                        "blob": {
+                            "enabled": true
+                        },
+                        "file": {
+                            "enabled": true
+                        }
+                    }
+                },
+                "networkAcls": {
+                    "bypass": "AzureServices",
+                    "defaultAction": "Allow",
+                    "ipRules": [],
+                    "virtualNetworkRules": []
+                },
+                "supportsHttpsTrafficOnly": true
+            },
+            "type": "Microsoft.Storage/storageAccounts"
```

Comment:
move this to the top of the resource definition - need to do it for all resources in all files

Reply:
Done